### PR TITLE
Add: Backend integration and refactoring

### PR DESCRIPTION
**Pull request by abdullahranginwala**
# Description

The backend was earlier written to work in the renderer process. Hence, it had access to the DOM elements. I have reconfigured the backend such that opencv.js now works in Node.js. I have also setup IPC between the react and node.js process to efficiently handle the connection between the backend and the front-end.

Since the backend was a major chunk of the application, and majority of the functionality comes from the backend, I couldn't break up the PR. However, I have added multiple commits for better maintainability. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

Also, include screenshots for verification and reviewing purpose.

https://github.com/scorelab/imagelab/assets/19731056/a857b603-56ca-4297-a99c-0d4cb3b6222e

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: imagelab_react/src/imagelab-block.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ Blockly.defineBlocksWithJsonArray([
         {
           name: "file_browse",
           type: "field_image",
-          src: "imagelab_demo/src/image-plus.png",
+          src: "./image-plus.png",
```
Comment (SahanDisa) [inline imagelab_react/src/imagelab-block.js:14]: why you have to change the path? @abdullahranginwala

File: imagelab_react/src/App.js [hunk 2]
```diff
@@ -2,30 +2,22 @@ import './App.css';
...
-  }
-
-  const handleReset = () => {
-    const shouldReset = window.confirm('Are you sure you want to reset the workspace?');
-    if (shouldReset) {
-      Blockly.mainWorkspace.clear();
-    }
+  const start = () => {
+    const topBlock = Blockly.getMainWorkspace().getTopBlocks()[0];
+    run(topBlock);
```
Comment (SahanDisa) [inline imagelab_react/src/App.js:20]: what happened to the other two functions? @abdullahranginwala handleRedo and handleReset

File: imagelab_react/src/utils/main.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      // Send message to the parent window
+      window.postMessage({ type: 'imageProcessed', canvasImageData }, '*');
+  
+      console.log('Image stored in localStorage');
+    };
+  
+    image.src = base64Image;
+  }
+
+export function undo() {
```
Comment (SahanDisa) [inline imagelab_react/src/utils/main.js:77]: is here you define the undo and reset?

Comment (SahanDisa) [reply imagelab_react/src/utils/main.js:77]: please add a comment on app.js if so

**Review by SahanDisa — COMMENTED** [review]
@abdullahranginwala can you add your answers for the reviews that I raised

File: imagelab_react/src/imagelab-block.js (hunk 1 above)
Comment (abdullahranginwala) [reply imagelab_react/src/imagelab-block.js:14]: This path was earlier used for the demo project I had made. When I shifted the project into imagelab and renamed the project to imagelab_react, this path was remained unchanged. It is now rendering the correct image.

File: imagelab_react/src/utils/main.js (hunk 3 above)
Comment (abdullahranginwala) [reply imagelab_react/src/utils/main.js:77]: Yes, I shifted the undo, redo and reset functions into the utils folder and I am importing into App.js directly from there. I'll add a comment there.

File: imagelab_react/src/imagelab-block.js (hunk 1 above)
Comment (SahanDisa) [reply imagelab_react/src/imagelab-block.js:14]: alright, got it

File: imagelab_react/src/utils/main.js (hunk 3 above)
Comment (SahanDisa) [reply imagelab_react/src/utils/main.js:77]: thanks for the clarification

**Review by SahanDisa — APPROVED** [review]

**Pull request merged**